examples/Planning/ForecastReach.cs

```diff
@@ -0,0 +1,313 @@
+            {
+                new DeviceInfo() { Type = Device.Desktop },
+                new DeviceInfo() { Type = Device.Mobile },
+                new DeviceInfo() { Type = Device.Tablet }
+            };
+
+            Targeting targeting = new Targeting()
+            {
+                PlannableLocationId = locationId,
+                AgeRange = ReachPlanAgeRange.AgeRange1865Up,
+
+            };
+            targeting.Genders.AddRange(genders);
+            targeting.Devices.AddRange(devices);
+
+            GenerateReachForecastRequest request = new GenerateReachForecastRequest()
+            {
+                CustomerId = customerId,
+                CurrencyCode = currencyCode,
+                CampaignDuration = duration,
+                Targeting = targeting,
+                MinEffectiveFrequency = 3
+            };
+
+            request.PlannedProducts.AddRange(productMix);
+
+            return request;
+        }
+
+        /// <summary>
+        /// Pulls and prints the reach curve for the given request.
+        /// </summary>
+        /// <param name="reachPlanService">Instance of Reach Plan Service client.</param>
+        /// <param name="request">An already-populated reach curve request.</param>
+        public void PullReachCurve(ReachPlanServiceClient reachPlanService, GenerateReachForecastRequest request)
+        {
+            GenerateReachForecastResponse response = reachPlanService.GenerateReachForecast(request);
+            Console.WriteLine("Reach curve output:");
+            Console.WriteLine(
+                "Currency,\tCost,\tOn-Target Reach,\tOn-Target Imprs,\tTotal Reach,\tTotal Imprs,\tProducts");
+            foreach (ReachForecast point in response.ReachCurve.ReachForecasts)
+            {
+                Console.Write($"{request.CurrencyCode},\t");
+                Console.Write($"{point.CostMicros / 1.0e6},\t");
+                Console.Write($"{point.Forecast.OnTargetReach},\t");
+                Console.Write($"{point.Forecast.OnTargetImpressions},\t");
+                Console.Write($"{point.Forecast.TotalReach},\t");
+                Console.Write($"{point.Forecast.TotalImpressions},\t");
+                Console.Write($"\"[");
+                foreach (ProductAllocation product in point.ForecastedProductAllocations)
+                {
+                    Console.Write($"(Product: {product.PlannableProductCode}, ");
+                    Console.Write($"Budget: {product.BudgetMicros / 1.0e6}), ");
+                }
+                Console.WriteLine($"]\"");
+            }
+        }
+
+        /// <summary>
+        /// Pulls a forecast for a manually allocated product mix.
+        /// </summary>
+        /// <param name="reachPlanService">Instance of Reach Plan Service client.</param>
+        /// <param name="customerId">The customer ID for the reach forecast.</param>
+        /// <param name="locationId">Location ID from somewhere returned by ListPlannableLocations.</param>
```

Comment:
This cannot be "somewhere", nor can you expect developers to always call ListPlannableLocations.

Instead, change this to:

Location IDs to plan for. You can get a valid location ID from 
https://developers.google.com/adwords/api/docs/appendix/geotargeting or by making a call to <see ...ListPlannableLocations/> method.

Reply:
Fixed throughout.